src/devices/esp8266.c

```diff
@@ -123,6 +123,32 @@ static void ipd_urc_cb(char *msg)
 		state.hooks.data_received_cb(chan_id, len, data);
 }
 
+static void wifi_action_callback(const char* msg)
+{
+	/* Check for the hook */
+	if (!state.hooks.client_state_changed_cb)
+		return;
+
+	enum client_action action;
+	/* Look at our 5th character to determin our action */
+	switch (msg[5]) {
```

Comment:
It does do 6 characters though because you are not counting the null byte.
That is implicit in that it must be there.  Hence why this is safe

On Tue, Aug 30, 2016, 17:13 Brent Picasso <notifications@github.com> wrote:

> In src/devices/esp8266.c
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/834#discussion_r76889883>
> :
>
> > @@ -123,6 +123,32 @@ static void ipd_urc_cb(char *msg)
> >  		state.hooks.data_received_cb(chan_id, len, data);
> >  }
> >
> > +static void wifi_action_callback(const char* msg)
> > +{
> > +	/* Check for the hook */
> > +	if (!state.hooks.client_state_changed_cb)
> > +		return;
> > +
> > +	enum client_action action;
> > +	/* Look at our 5th character to determin our action */
> > +	switch (msg[5]) {
>
> A bit pedantic but worth pointing out; the strncmp() used in the function
> that calls this function only checks for the string "WIFI " - it doesn't
> check if the string is at least 6 characters to check for the start of that
> next word.
>
> if (strncmp(msg, "WIFI ", 5) == 0) {
>
>    - wifi_action_callback(msg);
>    - return true;
>    - }
>
> What would happen is it would land on the null terminator - technically
> safe but a more explicit check would be to check for minimum length too.
>
> Too bad strncmp() doesn't provide wildcards! :)
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/834/files/8ef5c56046a2a89e6670200efb3d33ad30824727#r76889883>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIoz5GjPUyeUUgYi66qe43rcl1_fdHCks5qlKsIgaJpZM4JtS2Y>
> .
>